src/pages/_app.tsx

```diff
@@ -55,11 +56,13 @@ function MyApp({ Component, pageProps, session }: AppProps & CustomProps) {
                 <CacheProvider value={customCache(theme)}>
                     <ThemeProvider theme={theme}>
                         <OverlayProvider>
-                            <Layout pagePath={pagePath}>
-                                <ErrorBoundary>
-                                    <Component {...pageProps} />
-                                </ErrorBoundary>
-                            </Layout>
+                            <RequestContentModalProvider>
+                                <Layout pagePath={pagePath}>
+                                    <ErrorBoundary>
+                                        <Component {...pageProps} />
+                                    </ErrorBoundary>
+                                </Layout>
+                            </RequestContentModalProvider>
```

Comment:
Is there any overhead to wrapping the whole app with this? Just because there are a lot of routes that don't have the request content functionality, and I'm wondering if it might make more sense to just wrap the page templates that do have that functionality instead?

Reply:
thats a great point, ill look into moving it elsewhere

Reply:
so i looked into this a little bit more...the issue is that `<ContentType />` is returned separately 7 times, plus adding the other usages, I would probably have to wrap `<RequestContentModalProvider />` 10+ times, which I would say mitigates the purpose of trying to use this context. If we think using it in context doesn't provide any value to the app, I can revert it back to local state no problem